Add Greenland holidays

Klintrup opened the pull request:
## Proposed change

Add Greenland holidays with Greenlandic, en_US and Danish translations

## Type of change

- [x] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

Klintrup commented:
I could not complete the pre-commit/push rules without adding the changes to the CH.po files, if you want I can remove them from the PR and you can push these in another PR.

PPsyrius commented:
I find "Epiphany" (Jan 6) being listed as Public Holidays too on another source: https://visitgreenland.com/practical-questions/#toggle-id-33 . Can you confirm for me how recent was this added/removed from their list?

```
A LIST OF PUBLIC HOLIDAYS IN GREENLAND
- 1 Jan New Year’s Day.
- 6 Jan Epiphany (generally a half day of holiday)
- Monday-Thursday (the Thursday before Easter Sunday)
- Good Friday (The Friday before Easter Sunday)
- Easter Sunday (March/April)
- Easter Monday (The day after Easter Sunday)
- General Prayer Day (the fourth Friday after Easter)
- 1 May Labor Day (generally a half day of holiday)
- Ascension Day (40 days after Easter).
- Whit Sunday (seven weeks after Easter).
- Whit Monday (same as above + 1 day)
- [21 Jun National Day.](https://visitgreenland.com/about-greenland/national-day-greenland/)
- 24 Dec Christmas Eve
- 25 Dec First Day of Christmas.
- 26 Dec second Day of Christmas.
- 31 Dec New Year’s Eve.
```

Klintrup commented:
> I find "Epiphany" (Jan 6) being listed as Public Holidays too on another source: https://visitgreenland.com/practical-questions/#toggle-id-33 . Can you confirm for me how recent was this added/removed from their list?
> 
> ```
> A LIST OF PUBLIC HOLIDAYS IN GREENLAND
> - 1 Jan New Year’s Day.
> - 6 Jan Epiphany (generally a half day of holiday)
> - Monday-Thursday (the Thursday before Easter Sunday)
> - Good Friday (The Friday before Easter Sunday)
> - Easter Sunday (March/April)
> - Easter Monday (The day after Easter Sunday)
> - General Prayer Day (the fourth Friday after Easter)
> - 1 May Labor Day (generally a half day of holiday)
> - Ascension Day (40 days after Easter).
> - Whit Sunday (seven weeks after Easter).
> - Whit Monday (same as above + 1 day)
> - [21 Jun National Day.](https://visitgreenland.com/about-greenland/national-day-greenland/)
> - 24 Dec Christmas Eve
> - 25 Dec First Day of Christmas.
> - 26 Dec second Day of Christmas.
> - 31 Dec New Year’s Eve.
> ```

The "Half day" are not official holidays but some union agreements and schools will let parents and kids off on those days, but there is no country-wide agreement on those.

I've found a better list here: https://www.timeanddate.com/holidays/greenland/2024 and will add it as reference along with the Epiphany as an optional holiday.

arkid15r commented on holidays/countries/greenland.py:
```diff
@@ -0,0 +1,96 @@
+        self._add_christmas_day_two(tr("Juullip aappaa"))
+
+    def _populate_optional_holidays(self):
+        # Epiphany
```
```suggestion
        # Epiphany.
```

Klintrup replied:
Implemented

KJhellico reviewed (commented):
@Klintrup, it's just perfect PR! Thank you!

arkid15r reviewed (commented):
@Klintrup thank you for adding Greenland holidays and also for doing it in a clean code fashion!

>I could not complete the pre-commit/push rules without adding the changes to the CH.po files, if you want I can remove them from the PR and you can push these in another PR.

Right, this wasn't related to your PR and has been fixed. I updated the PR's code and tests look green.

Optional: would you be willing to update `AUTHORS` file?

🇬🇱 LGTM

@KJhellico @PPsyrius any comments/suggestions? I hope to include these changes into Monday's v0.50

Thank you!

Klintrup commented:
> @Klintrup thank you for adding Greenland holidays and also for doing it in a clean code fashion!
> 
> > I could not complete the pre-commit/push rules without adding the changes to the CH.po files, if you want I can remove them from the PR and you can push these in another PR.
> 
> Right, this wasn't related to your PR and has been fixed. I updated the PR's code and tests look green.
> 
> Optional: would you be willing to update `AUTHORS` file?
> 
> 🇬🇱 LGTM
> 
> @KJhellico @PPsyrius any comments/suggestions? I hope to include these changes into Monday's v0.50
> 
> Thank you!

Thanks for the fix regarding CH.po

I updated the AUTHORS file as requested and implemented your proposed fix to keep the formatting uniform.
